research/object_detection/dataset_tools/create_ava_tf_record.py

```diff
@@ -0,0 +1,439 @@
+  return tf.train.FeatureList(feature=value)
+
+class Ava(object):
+  """Generates and loads the Kinetics data set."""
+
+  def __init__(self, path_to_output_dir, path_to_data_download):
+    if not path_to_output_dir:
+      raise ValueError("You must supply the path to the data directory.")
+    self.path_to_data_download = path_to_data_download
+    self.path_to_output_dir = path_to_output_dir
+
+  def generate_examples(self,
+                        splits_to_process="train,val,test",
+                        video_path_format_string=None,
+                        download_labels_for_map=True,
+                        seconds_per_sequence=10,
+                        hop_between_sequences=10):
+    """Downloads data and generates sharded TFRecords.
+
+    Downloads the data files, generates metadata, and processes the metadata
+    with MediaPipe to produce tf.SequenceExamples for training. The resulting
+    files can be read with as_dataset(). After running this function the
+    original data files can be deleted.
+
+    Args:
+      splits_to_process: csv string of which splits to process. Allows providing
+        a custom CSV with the CSV flag. The original data is still downloaded
+        to generate the label_map.
+      video_path_format_string: The format string for the path to local files.
+      download_labels_for_map: If true, download the annotations to create the
+        label map.
+      seconds_per_sequence: The length of each sequence, in seconds.
+      hop_between_sequences: The gap between the centers of
+      successive sequences.
+    """
+    logging.info("Downloading data.")
+    download_output = self._download_data(download_labels_for_map)
+    for key in splits_to_process.split(","):
+      logging.info("Generating metadata for split: %s", key)
+      all_metadata = list(self._generate_metadata(
+          download_output[0][key][0], download_output[0][key][1],
+          download_output[1], seconds_per_sequence, hop_between_sequences,
+          video_path_format_string))
+      logging.info("An example of the metadata: ")
+      logging.info(all_metadata[0])
+      random.seed(47)
+      random.shuffle(all_metadata)
+      shards = SPLITS[key]["shards"]
+      shard_names = [os.path.join(
+          self.path_to_output_dir, FILEPATTERN % key + "-%05d-of-%05d" % (
+              i, shards)) for i in range(shards)]
+      writers = [tf.io.TFRecordWriter(shard_name) for shard_name in shard_names]
+      with _close_on_exit(writers) as writers:
+        for i, seq_ex in enumerate(all_metadata):
+          writers[i % len(writers)].write(seq_ex.SerializeToString())
+    logging.info("Data extraction complete.")
+
+  def _generate_metadata(self, annotation_file, excluded_file, label_map,
+                         seconds_per_sequence, hop_between_sequences,
+                         video_path_format_string):
+    """For each row in the annotation CSV, generates the corresponding metadata
+
```

Comment:
Please expand the description to indicate the set of frames you pick for each sequence example.